csr.py

```
@@ -166,100 +121,104 @@
+                             raise KeyError("Common Name (CN) cannot be empty.")
+                        raise KeyError(f"Required field {field} cannot be empty.")
+                    valid[field] = ""
+                    continue
                 
                 field_value = form_values[field]
-                
-                # Validate field length limits
+
+                # Normalize string fields that might contain international characters or varied representations
+                if field in ['CN', 'O', 'OU', 'L', 'ST', 'subjectAltNames'] and isinstance(field_value, str):
+                    field_value = unicodedata.normalize('NFC', field_value.strip())
```

Ruff (0.11.9)

Undefined name `unicodedata`

(F821)